### ISSUE TITLE
fix(training): preserve row pairing in flow finetune loss

### ISSUE THREAD

**Issue by ktinubu** [tinaudio/synth-setter#3390]
## Problem

`VSTFlowFinetuneModule._train_step` multiplies a `(batch,)` row-loss tensor by `(batch, 1)` time weights. Broadcasting produces a `(batch, batch)` outer product, so each row loss is paired with every row weight instead of its own.

## Expected

Reduce the squared error with `keepdim=True` (or otherwise retain `(batch, 1)`) before applying time weights, matching the base flow module row-pairing contract.

## Reproduction

Use a two-row finetune batch with unequal row errors and weights. The current scalar equals the outer-product average rather than the mean of the two paired weighted losses.

Found while adapting `TrainStepOutputs` for #3389; out of scope for that prerequisite.

**Comment by ktinubu** [comment]
`[from W7]` **claimed** — verified unassigned, no branch, no worktree, no open PR referencing it.

Reproduced by inspection at `0f81e9c9d6`: `src/synth_setter/models/vst_flow_finetune_module.py:660` is

```python
loss = (squared_flow_error.mean(dim=-1) * w).mean()
```

`squared_flow_error.mean(dim=-1)` is `(batch,)` and `w` is `(batch, 1)`, so the product is the `(batch, batch)` outer product — exactly as filed. The base module (`vst_flow_matching_module.py:1030`) already does `squared_flow_error.mean(dim=-1, keepdim=True)` before `* w`, so the finetune override diverges from the contract it inherits.

Working in `wt/w7-3390` on `internal-fix/3390-finetune-row-pairing`, based on `0f81e9c9d6`. Test-first: a two-row batch with unequal row errors and unequal time weights, asserting the scalar equals the paired weighted mean and not the outer-product mean.
